pkg/config/validation/validation.go

```diff
@@ -524,6 +524,12 @@ func validateServer(server *networking.Server) (v Validation) {
 				v = appendValidation(v, fmt.Errorf("server cannot have TLS settings for non HTTPS/TLS ports"))
 			}
 		}
+
+		if p.IsTLS() {
```

Comment:
I am not sure this is right, we can set Server.Tls but have port named HTTP

You can see `MergeGateways`